### PR TITLE
bugfix (TCOMP-364) : upgrade to jetty 9 to have servlet 3.1 dependency

### PR DESCRIPTION
**Pull request by vjaouen**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
The oauth component use servlet 3.0 dependency.

**What is the new behavior?**
The oauth component use servlet 3.1 dependency.

**Does this PR introduce a breaking change?**
- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
unit test added

### CODE REVIEW

File: components-common-oauth/src/main/java/org/talend/components/common/oauth/util/HttpsService.java [hunk 1]
```diff
@@ -33,14 +32,32 @@
...
 
-        SslSelectChannelConnector ssl_connector = new SslSelectChannelConnector();
-        ssl_connector.setHost(host);
-        ssl_connector.setPort(port);
-        SslContextFactory cf = ssl_connector.getSslContextFactory();
-        cf.setKeyStorePath(HttpsService.class.getResource("sslkey").toString());
-        cf.setKeyStorePassword("talend");
-        cf.setKeyManagerPassword("talend");
-        server.addConnector(ssl_connector);
+        HttpConfiguration httpConfig = new HttpConfiguration();
```
Comment (sebhoss) [inline components-common-oauth/src/main/java/org/talend/components/common/oauth/util/HttpsService.java]: `httpConfig` is only used by the copy-constructor in line 47. How about we move lines 36 + 37 down and configure `httpsConfig` directly? We need one variable less and the configuration of `httpsConfig` happens all in one place.

Comment (vjaouen) [inline components-common-oauth/src/main/java/org/talend/components/common/oauth/util/HttpsService.java]: I agree with you.
Change request done !

File: components-common-oauth/pom.xml [hunk 2]
```diff
@@ -22,7 +22,7 @@
 		<dependency>
 			<groupId>org.eclipse.jetty</groupId>
 			<artifactId>jetty-server</artifactId>
-			<version>8.1.17.v20150415</version>
+			<version>9.3.14.v20161028</version>
```
Comment (dkulp) [inline components-common-oauth/pom.xml]: Jetty 9.3 requires Java 8, do we have the OK to proceed without Java7 support?

Comment (vjaouen) [inline components-common-oauth/pom.xml]: Good point !
I downgraded to jetty 9.2.
Thanks.

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/860/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/860/)

**Comment by gonchar-ivan** [comment]
Will Jetty 9.2 compatible with Jetty dependency which comes from Spring/Spring Boot.

The idea is to align version of Jetty (and other dependencies) across modules in TCOMP (components repo and daikon)

**Comment by vjaouen** [comment]
Jetty 9.2 is compatible with spring boot 1.4.1.

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/861/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/861/)

**Comment by sgandon** [comment]
There are some osgi test failing, I'll look into it right now.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/910/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/910/)

**Pull request merged**